platform/mds/mds/src/main/java/org/motechproject/mds/annotations/Discriminated.java

```diff
@@ -0,0 +1,13 @@
+package org.motechproject.mds.annotations;
+
+import javax.jdo.annotations.IdentityType;
+import javax.jdo.annotations.PersistenceCapable;
+import java.lang.annotation.*;
+
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@PersistenceCapable(identityType = IdentityType.DATASTORE, detachable = "true")
+public @interface Discriminated {
```

Comment:
Surprising: ". . .wondering about the purpose. . ."

Here's my best shot at clearing that up:  No I would not agree that the purpose is the UI behavior, though I could see how many (even most) might first be perceive the feature that way.

Also surprising: "Why not use JDO _ annotations?".

Maybe this helps? My answer to this is the same as my answer to:
  - Why @Entity?  Why not @PersistenceCapable?
  - Why have an Entity table?  Why not ask the database?
  - Why use MDS?  Why not use JDO?

Think of @Discriminated as the implementation of a datatype.  Originally there was a separate JIRA for it: https://applab.atlassian.net/browse/MOTECH-1768 though in practice it was not practical to develop separately.

Similarly, think of MOTECH-1649 as being Data Browser support for @Discriminated data.